### PR TITLE
Math typesetting support with KaTeX

### PR DESCRIPTION
**Pull request by mr-chrome**
In order to implement math typesetting in Zerm, I took inspiration from the work I did with my
previous pelican blog; moreover, I saw that the Zola DeepThought theme did something similar
relying on the KaTeX CDN. Instead, I made a dedicated directory in static called `katex` and 
put the minimal assets (js, css and fonts) right in there.

The assets are called **only** when a page has the `math` field stated as `true` in the frontmatter
and if the `enable_katex` field is declared as `true` in the main config file.
I tested this for about a week with a bunch of different articles and it seems to work just fine.
If you want to see the function in action you can check the README of the master branch: I 
linked a working demo right there.

### CODE REVIEW

**Comment by ejmg** [comment]
thank you so much for the PR and sorry for the late response on my part. a few things:
- can you remove your changes to README?
- furthermore, please advertise the existence of this feature in the README along with explaining how this doesn't add a performance hit for anyone who does not opt into using it. You could also just limit this explanation to the Config file.

@adworacz, if you have the chance, can you sanity check that this PR won't add overhead for anyone who doesn't opt into the use of KaTeX?

File: README.md [hunk 1]
```diff
@@ -27,6 +23,10 @@ emphasize minimalism even more. Some of them are as follows:
 Some of these might be added later and [PR's are always
 welcomed](https://github.com/ejmg/zerm/pulls).
 
+## differences from ejmg port
+
+- math typesetting support with KaTeX.
```
Comment (adworacz) [inline README.md:28]: This is a merge request *to* the ejmg port, so all of this "fork" / "differences" talk shouldn't be pulled in.

File: templates/page.html [hunk 2]
```diff
@@ -1,5 +1,12 @@
+{% import "macros/head.html" as head -%}
 {% extends "index.html" -%}
 
+{%- block math -%}
+  {% if page.extra.math %}
+  {{ head::katex() }}
+  {% endif %}
```
Comment (adworacz) [inline templates/page.html:7]: @ejmg There's two levels of checks that are preventing this katex addition from adding to total download size for anyone who doesn't explicitly turn on katex support.

1. This is turned on a per-page level, meaning that even when it's enabled for a whole site, it will only add to the total download size for the specific pages that enable it.
2. The check on line 46 of `head.html` enables/disables the pertinent resources across the entire site *even if* a specific page attempted to turn it on.

In short, this should be a pretty safe change re: download size, and won't impact other users of the theme, aside from add unused files to their `static/assets` directory and taking up (minimal) room on their server.

File: static/assets/katex/katex.min.css [hunk 3]
```diff
@@ -0,0 +1 @@
...y:before{content:""}.katex .hide-tail{width:100%;position:relative;overflow:hidden}.katex .halfarrow-left{position:absolute;left:0;width:50.2%;overflow:hidden}.katex .halfarrow-right{position:absolute;right:0;width:50.2%;overflow:hidden}.katex .brace-left{position:absolute;left:0;width:25.1%;overflow:hidden}.katex .brace-center{position:absolute;left:25%;width:50%;overflow:hidden}.katex .brace-right{position:absolute;right:0;width:25.1%;overflow:hidden}.katex .x-arrow-pad{padding:0 .5em}.katex .mover,.katex .munder,.katex .x-arrow{text-align:center}.katex .boxpad{padding:0 .3em}.katex .fbox,.katex .fcolorbox{box-sizing:border-box;border:.04em solid}.katex .cancel-pad{padding:0 .2em}.katex .cancel-lap{margin-left:-.2em;margin-right:-.2em}.katex .sout{border-bottom-style:solid;border-bottom-width:.08em}.katex-display{display:block;margin:1em 0;text-align:center}.katex-display>.katex{display:block;text-align:center;white-space:nowrap}.katex-display>.katex>.katex-html{display:block;position:relative}.katex-display>.katex>.katex-html>.tag{position:absolute;right:0}.katex-display.leqno>.katex>.katex-html>.tag{left:0;right:auto}.katex-display.fleqn>.katex{text-align:left;padding-left:2em}
```
Comment (adworacz) [inline static/assets/katex/katex.min.css:1]: This is referencing fonts (`KaTeX_Caligraphic-Regular`, `fonts/KaTeX_SansSerif-Regular.woff2` etc), that aren't included in this PR. If they are unnecessary, then please remove them from the CSS.

**Review by adworacz — CHANGES_REQUESTED** [review]
Sorry, @ejmg just noticed this PR. Took a look and added some comments below.

**Comment by mr-chrome** [comment]
Sorry for the long absence: a month ago I was studying for a exam and only now I was awaken by this new reply. I'm going to work at this changes as soon as possible

File: static/assets/katex/katex.min.css (hunk 3 above)
Comment (mr-chrome) [reply static/assets/katex/katex.min.css:1]: I didn't touch this minified CSS: fonts can be requested by the user and takes up very minimal space on server. I just forgot to add some font during the developing phase, but all fonts are up now.

File: templates/page.html (hunk 2 above)
Comment (mr-chrome) [reply templates/page.html:7]: I think it's better leaving per-page level control instead of the global one. Now the math works by default on every article with `math=true` in the frontmatter (`[extra]` section, of course). Other articles aren't touched in any way from this feature.

File: README.md (hunk 1 above)
Comment (mr-chrome) [reply README.md]: My bad; now it should be okay: I added a paragraph which explains the new feature and leaved the other parts of the README as before.

**Review by ejmg — CHANGES_REQUESTED** [review]
it's a single if check which rips out katex entirely if opted out of, i think it's better to include this as @adworacz suggested.

**Comment by ejmg** [comment]
beyond that, this is a great PR and thank you for putting the work into it

**Comment by mr-chrome** [comment]
So @ejmg you think it's better to make it as I did at the start (with the check in the config file too)? 
I can put it on again, of course, but I thought @adworacz  suggested me to remove that, so I did it. Maybe I'm misunderstanding something?

**Comment by adworacz** [comment]
Let me clarify - my comments weren’t suggesting to remove any checks. 

Those comments were addressing @ejmg‘s request to audit the code to ensure it wouldn’t cause any additional overhead (bandwidth usage) for any users who leave this feature disabled. Specifically, I was commenting that your two-tier config check made it a very safe change.

**Comment by mr-chrome** [comment]
@adworacz I see, this is my very first PR on an external project and I totally misunderstood your point, but it's all clear now. I will restore the check again ASAP, thanks

**Comment by ejmg** [comment]
it did not help that i misread GitHub's review as edits and misinterpreted the discussion further. just ping me again when it's ready @mr-chrome  and thank you for the clarification yet again Austin.

**Comment by mr-chrome** [comment]
@ejmg I think it's fine now, but I'm not able to resolve the conflict on the config file. Maybe I just cannot because it's up to you, truth is I'm not that confident with PRs, so I don't want to mess it up with other useless commits. If there's something I can do, just let me know. Thanks for your work

**Comment by ejmg** [comment]
Don't feel bad about the merge stuff, it can be very confusing for anyone who doesn't regularly work with Git + GitHub/GitLab/etc.

The conflict should be fixed now, so pulling the PR branch should get the branch back to normal on your machine *if* you have more work you want to add. Or do you consider this PR done?

**Review by ejmg — APPROVED** [review]

**Comment by mr-chrome** [comment]
@ejmg I made other significant edits in other branches which are useful for my website and I may apply something in the future, but I consider this particular PR done, yes.
